src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,231 @@
+  static Gson gson = new Gson();
+
+  public static boolean validateSalesforceToBQRecordValues(String objectName, String targetTable) throws
+    IOException, InterruptedException {
+    String id = SalesforceClient.id;
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateBQToSalesforceRecordValues(String targetTable, String objectName) throws
+    IOException, InterruptedException {
+    SalesforceClient.queryObjectID(objectName);
+    String id = SalesforceClient.id;
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateSalesforceMultiObjectToBQ(String objectName) throws IOException,
+    InterruptedException {
+    String id = SalesforceClient.id;
+    String targetTable = getTableByName();
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+  }
+
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows) throws IOException,
+    InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  public static TableResult getTableNamesFromDataSet() throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT table_name FROM `" + projectId + "." + dataset + "`.INFORMATION_SCHEMA.TABLES ";
+
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  public static String getTableByName() throws IOException, InterruptedException {
+    String tableName = PluginPropertyUtils.pluginProp("bqTargetTable");
```

Comment:
I believe this method is used for multi object validation. Try to handle few edge cases here. 

1. Check if table name is same as Object name, if yes add those 2 tables names in the list .In case your list don't have any table fail the test here only with a message that database don't have any table or something.

2. if you are using 2 tables in the source you need to 2 tables from the list, so check for this size once you have retrived the tables if size is not same fail the test.

Reply:
done